packages/protocol/lib/memview-sol/.gitignore

```diff
@@ -0,0 +1,6 @@
+node_modules/
```

Comment:
Are the files in `lib/memview-sol/` all from the `foundry` installation? Is it typical to commit these to a repo using a `foundry` package, or should this be all ignored?

Reply:
> Are the files in lib/memview-sol/ all from the foundry installation

Yes.

> Is it typical to commit these to a repo using a foundry package, or should this be all ignored?

I have no idea. I only discovered foundry during the CI failure and have never used it before. I would have preferred to not commit these files, but since I couldn't find an existing build step where new dependencies are installed this seemed like the least invasive way to make it work.

I could also add more foundry ignores so that this is not even necessary, since the code is tested with the TS test runner.

As mentioned in the PR description I was hoping for some advice on the foundry situation:
> I'm not sure if it was correct to use forge install to install the memview package. If it was, I wonder how I should proceed with the [tslint error](https://app.circleci.com/pipelines/github/celo-org/celo-monorepo/44172/workflows/fb258e07-ea09-47ed-aca4-536d9816df44/jobs/584188/parallel-runs/0/steps/0-103). Manually editing external dependencies to fix lint errors seems wrong, since it will make it harder to update dependencies.

Reply:
Looking at some of the other foundry packages (openzeppelin-contracts, solidity-bytes-utils, etc.), their directories in the repo are empty.

> but since I couldn't find an existing build step where new dependencies are installed this seemed like the least invasive way to make it work

Yep, also not seeing how this happens. Not very familiar with foundry myself though. Are there missing steps from our build scripts that should be added? Do you know how/when the other foundry packages are installed (like openzeppelin-contracts)?

That said, to your point of whether we should even be using foundry for this, I think there is an argument to be made for just copying the files over to our repo and maintaining them ourselves rather than relying on an external package.

Reply:
@bowd You added the foundry usage. Can you give us some hints on this?